src/main/java/org/embulk/input/marketo/delegate/CampaignInputPlugin.java

```diff
@@ -0,0 +1,66 @@
+package org.embulk.input.marketo.delegate;
+
+import com.google.common.collect.FluentIterable;
+import org.embulk.base.restclient.ServiceResponseMapper;
+import org.embulk.base.restclient.jackson.JacksonServiceResponseMapper;
+import org.embulk.base.restclient.record.RecordImporter;
+import org.embulk.base.restclient.record.ServiceRecord;
+import org.embulk.base.restclient.record.ValueLocator;
+import org.embulk.config.TaskReport;
+import org.embulk.input.marketo.MarketoService;
+import org.embulk.input.marketo.MarketoServiceImpl;
+import org.embulk.input.marketo.MarketoUtils;
+import org.embulk.input.marketo.rest.MarketoRestClient;
+import org.embulk.spi.Exec;
+import org.embulk.spi.PageBuilder;
+import org.embulk.spi.type.Types;
+
+/**
+ * Input plugin use to import Campaign
+ * Created by tai.khuu on 9/18/17.
+ */
+public class CampaignInputPlugin extends MarketoBaseInputPluginDelegate<CampaignInputPlugin.PluginTask>
+{
+    public interface PluginTask extends MarketoBaseInputPluginDelegate.PluginTask
+    {
+    }
+
+    public CampaignInputPlugin()
+    {
+    }
+
+    @Override
+    public TaskReport ingestServiceData(PluginTask task, RecordImporter recordImporter, int taskIndex, PageBuilder pageBuilder)
+    {
+        try (MarketoRestClient marketoRestClient = createMarketoRestClient(task)) {
```

Comment:
This method seems to be duplicated (only difference in service call) with the ones in `LeadWithListInputPlugin` & `LeadWithProgramInputPlugin` which is a good sign to refactor it out into a base class.